projects/packages/wp-js-data-sync/src/endpoints/class-endpoint.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Register and handle REST API Endpoints for each data sync entry.
+ *
+ * @package automattic/jetpack-wp-js-data-sync
+ */
+
+// phpcs:disable Squiz.Commenting.FunctionComment.MissingParamComment
+// phpcs:disable Squiz.Commenting.FunctionComment.MissingParamName
+// phpcs:disable Squiz.Commenting.FunctionComment.MissingParamTag
+// phpcs:disable Squiz.Commenting.FunctionComment.MissingReturn
+// phpcs:disable Generic.Commenting.DocComment.MissingShort
+// phpcs:disable Squiz.Commenting.FunctionComment.Missing
+// phpcs:disable Squiz.Commenting.ClassComment.Missing
+// phpcs:disable Squiz.Commenting.FileComment.Missing
+
+namespace Automattic\Jetpack\WP_JS_Data_Sync\Endpoints;
+
+use Automattic\Jetpack\WP_JS_Data_Sync\Data_Sync_Entry;
+
+class Endpoint {
+
+	/**
+	 * @var Data_Sync_Entry $entry - The data sync entry to register the endpoint for.
+	 */
+	private $entry;
+
+	/**
+	 * @var string $rest_namespace - The namespace for the REST API endpoint.
+	 */
+	private $rest_namespace;
+
+	/**
+	 * @var string $route - The route for the REST API endpoint.
+	 */
+	private $route;
+
+	/**
+	 * @var Authenticated_Nonce $nonce - The nonce for the REST API endpoint.
+	 */
+	private $nonce;
+
+	/**
+	 * @param string          $namespace - The namespace for the REST API endpoint.
+	 * @param string          $route     - The route for the REST API endpoint.
+	 * @param Data_Sync_Entry $entry     The data sync entry to register the endpoint for.
+	 */
+	public function __construct( $namespace, $route, $entry ) {
+		$this->entry          = $entry;
+		$this->rest_namespace = $namespace;
+		$this->route          = $route;
+		$this->nonce          = new Authenticated_Nonce( "{$namespace}_{$entry->key()}" );
+	}
+
+	public function register_rest_route() {
+		register_rest_route(
+			$this->rest_namespace,
+			$this->route,
+			array(
+				'methods'             => \WP_REST_Server::ALLMETHODS,
+				'callback'            => array( $this, 'handler' ),
+				'permission_callback' => array( $this, 'permissions' ),
+			)
+		);
+	}
+
+	/**
+	 * Route the request to the apropriate handler.
+	 *
+	 * @param \WP_REST_Request $request - The request object.
+	 */
+	public function handler( $request ) {
+		$methods = array(
+			'GET'    => 'handle_get',
+			'POST'   => 'handle_post',
+			'DELETE' => 'handle_delete',
+		);
```

Comment:
Missing support for [some methods](https://github.com/Automattic/jetpack/blob/73f8544e1259e87a399102aef5147a06e5056740/projects/js-packages/svelte-data-sync-client/src/API.ts#L4)?

Reply:
Yep. Removed them for now.